### ISSUE TITLE
OptimizerChainFactory refactoring broke "quality" config

### ISSUE THREAD

**Issue by glye** [spatie/image-optimizer#218]
CC @0xb4lint 

Hi, sorry if I misunderstand things. I'm looking at the "OptimizerChainFactory refactor with config processor" change from 1.7.3:
https://github.com/spatie/image-optimizer/commit/e74cdf113dc1f2ac5511415d2094bb69e99cb260

The problem occurs when the config in `OptimizerChainFactory::create($config)` contains a 'quality' entry, which it can, according to the code in `getOptimizers()` below. Previously, such a "generic" quality would be applied to all optimizers that support it.

The call to the new `self::configHasOptimizer($config)` results in config being returned verbatim, including the 'quality' entry:
https://github.com/spatie/image-optimizer/commit/e74cdf113dc1f2ac5511415d2094bb69e99cb260#diff-4f7d0501c10934549ecacfae146962d03a7dcacf1ffab4cadef7984a68409cdfR32

Then when this is instantiated, we get an attempt to call `new quality(...)` which fails with a ClassNotFoundErrror.
https://github.com/spatie/image-optimizer/commit/e74cdf113dc1f2ac5511415d2094bb69e99cb260#diff-4f7d0501c10934549ecacfae146962d03a7dcacf1ffab4cadef7984a68409cdfR21

In short: Since 1.7.3 it is no longer possible to specify `$config['quality']`, doing so results in a crash.
Thanks for reading.

**Comment by 0xb4lint** [comment]
Hi @glye!

You are very right, sorry for the breaking change.
I was able to reproduce the issue with a failing test: https://github.com/0xb4lint/image-optimizer/actions/runs/9098545560/job/25008988995

Working on a fix. 😉

**Comment by glye** [comment]
Thank you!
